Sources/Public/BacktraceCrashReporter.swift

```diff
@@ -52,19 +56,49 @@ extension BacktraceCrashReporter: CrashReporting {
         try reporter.enableAndReturnError()
     }
     
+    // This function retrieves, constructs, and sends the pending crash report
     func pendingCrashReport() throws -> BacktraceReport {
         let reportData = try reporter.loadPendingCrashReportDataAndReturnError()
         let attributes = (try? AttributesStorage.retrieve(fileName: BacktraceCrashReporter.crashName)) ?? [:]
-        // NOTE: - no attachments in crash reports
-        return try BacktraceReport(report: reportData, attributes: attributes, attachmentPaths: [])
+        let attachmentPaths = copiedFileAttachments.map {$0.path}
+        return try BacktraceReport(report: reportData, attributes: attributes, attachmentPaths: attachmentPaths)
     }
     
+    // This function is called to copy stored file attachments
+    // from pending crashes so that they are not overwritten by the
+    // new app session
+    func copyFileAttachmentsFromPendingCrashes() -> [URL] {
+        guard let cacheDirectoryUrl = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first else {
+            BacktraceLogger.error("Could not get cache directory URL")
+            return [URL]()
+        }
+        let attachments = (try? AttachmentsStorage.retrieve(fileName: BacktraceCrashReporter.crashName)) ?? [:]
+        var copiedFileAttachments = [URL]()
+        for attachment in attachments {
+            let fileManager = FileManager()
```

Comment:
I don't think it will work, cacheDirectoryUrl calls a static function of FileManager (`FileManager.default.urls`). Otherwise for our file operations below we need an actual FileManager instance.

Reply:
`FileManager.default` is your Filemanager instance.

Reply:
We don't use `FileManager.default` here anymore. I agree with your feedback that it's better to get this path directly from `AttachmentsStorage`.